node/follower_node_test.go

```diff
@@ -127,3 +147,68 @@ func TestDevModeWarning(t *testing.T) {
 	require.NotNil(t, foundEntry)
 	require.Contains(t, foundEntry.Message, "Follower running on a devMode network. Must submit txns to a different node.")
 }
+
+func TestSyncRoundWithRemake(t *testing.T) {
+	// Extend TestSyncRound to simulate starting and stopping the network
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+
+	maxAcctLookback := uint64(100)
+
+	followNode, tempDir := remakeableFollowNode(t, "", maxAcctLookback)
+	addBlock := func(round basics.Round) {
+		b := bookkeeping.Block{
+			BlockHeader: bookkeeping.BlockHeader{
+				GenesisHash: followNode.ledger.GenesisHash(),
+				Round:       round,
+				RewardsState: bookkeeping.RewardsState{
+					RewardsRate: 0,
+					RewardsPool: poolAddr,
+					FeeSink:     sinkAddr,
+				},
+			},
+		}
+		b.CurrentProtocol = protocol.ConsensusCurrentVersion
+		err := followNode.Ledger().AddBlock(b, agreement.Certificate{})
+		require.NoError(t, err)
+
+		status, err := followNode.Status()
+		require.NoError(t, err)
+		require.Equal(t, round, status.LastRound)
+	}
+
+	// Part I. redo TestSyncRound
+	// main differences are:
+	// * cfg.DisableNetworking = true
+	// * cfg.MaxAcctLookback = 100 (instead of 4)
+
+	addBlock(basics.Round(1))
+
+	dbRound := uint64(followNode.Ledger().DBRound())
+	// Sync Round should be initialized to the ledger's dbRound + 1
+	require.Equal(t, dbRound+1, followNode.GetSyncRound())
+	// Set a new sync round
+	require.NoError(t, followNode.SetSyncRound(dbRound+11))
+	// Ensure it is persisted
+	require.Equal(t, dbRound+11, followNode.GetSyncRound())
+	// Unset the sync round and make sure get returns 0
+	followNode.UnsetSyncRound()
+	require.Equal(t, uint64(0), followNode.GetSyncRound())
+
+	// Part II. fast forward and then remake the node
+
+	newRound := basics.Round(2 * maxAcctLookback)
+	for i := basics.Round(2); i <= newRound; i++ {
+		addBlock(i)
+	}
+
+	followNode, _ = remakeableFollowNode(t, tempDir, maxAcctLookback)
+	status, err := followNode.Status()
+	require.NoError(t, err)
+	require.Equal(t, newRound, status.LastRound)
+
+	// syncRound should be at
+	// newRound - maxAcctLookback + 1 = maxAcctLookback + 1
+	syncRound := followNode.GetSyncRound()
+	require.Equal(t, uint64(maxAcctLookback+1), syncRound)
```

Comment:
When reverting [this line](https://github.com/algorand/go-algorand/blob/7b74ce2ea16eef6161b8d572ff197127835da858/node/follower_node.go#L139) to the original
```go
err = node.SetSyncRound(uint64(node.Ledger().NextRound()))
```

I got the following test failure because the original **SyncRound** was coming back as 201 (the next round) instead of now expected 101:
```
--- FAIL: TestSyncRoundWithRemake (0.32s)
    follower_node_test.go:213: 
                Error Trace:    /Users/zeph/github/algorand/go-algorand/node/follower_node_test.go:213
                Error:          Not equal: 
                                expected: 0x65
                                actual  : 0xc9
                Test:           TestSyncRoundWithRemake
FAIL
exit status 1
FAIL    github.com/algorand/go-algorand/node    0.736s
```